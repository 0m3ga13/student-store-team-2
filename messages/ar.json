{
    "Index": {
        "Title": "المتجر الصغير",
        "Home": "الرئيسية",
        "Products": "المنتجات",
        "Orders": "الطلبيات",
        "State": "الحالة",
        "Buy": "شراء",
        "Borrow": "إعارة",
        "Description": "هذه المنصة هي مربحة للجميع: القرضاء يحققون دخلاً إضافيًا، والمقترضون يقللون التكاليف.",
        "Profile": "الحسـاب",
        "For Sale": "للبيع",
        "For Borrow": "للإعارة",
        "Sale & Borrow": "إعارة وشراء",
        "Day": "لليوم",
        "How It Works": "طريـقة العمل",
        "Hello": "مرحبا",
        "Sign out": "تسجيل الخروج",
        "Sign In": "تسجيل الدخول",
        "Login with Google": "سحل الدخول بحسابك",
        "Sign Up With Google": "سحل الدخول بحسابك"
    }
}
