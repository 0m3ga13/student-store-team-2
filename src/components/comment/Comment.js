"use client";
import Image from "next/image";
import React, { useState } from "react";

import InputReview from "./inputReview";
import StarRating from "../starRating/StarRating";

const Comment = ({ userData, comments, reply }) => {
    const [showReplyInputs, setShowReplyInputs] = useState({});
    const [showReplies, setShowReplies] = useState({});
    const [isOpen, setIsOpen] = useState(false);

    const [replyText, setReplyText] = useState({}); // State to manage the text of the reply input
    const handleInputChange = (e, commentId) => {
        const { value } = e.target;
        setReplyText({ ...replyText, [commentId]: value }); // Update the reply text for the specific comment
    };

    const postReply = (commentId) => {
        // Logic to post the reply using the replyText state value for a specific comment
        const newReply = {
            text: replyText[commentId] || "", // Retrieve the reply text for the specific comment
            // Add other necessary fields like user info, timestamp, etc.
        };

        // Assuming reply is a nested object where replies for each comment are stored under their respective commentIds
        const updatedReplies = {
            ...reply,
            [commentId]: {
                ...(reply[commentId] || {}),
                [new Date().getTime()]: newReply, // Using timestamp as the unique ID (not recommended in a real app)
            },
        };

        // Update state to add the new reply
        // This should typically involve updating state through props or context in a real application
        // For this example, assuming direct state update (not recommended in a real app)
        setReplyText({ ...replyText, [commentId]: "" }); // Clear the reply input field after posting the reply
    };

    const toggleReplies = (commentId) => {
        setShowReplies((prev) => ({
            ...prev,
            [commentId]: !prev[commentId],
        }));
    };

    const renderReplies = (replies, commentId) => {
        if (!replies || !showReplies[commentId]) return null;

        return Object.keys(replies).map((replyId) => {
            const reply = replies[replyId];

            return (
                <div key={replyId} className='border-b border-gray-300 pb-4'>
                    <div className='flex items-center space-x-4'>
                        <div className='w-12 h-12 rounded-full overflow-hidden'>
                            <Image
                                className='w-full h-full object-cover'
                                src={userData.profile_pic} // Assuming each comment has a user object with a profilePic field
                                alt='User'
                                width={500}
                                height={500}
                            />
                        </div>
                        <div>
                            <h2 className='text-sm font-semibold'>
                                {userData.firstName} {userData.lastName}
                            </h2>
                            <p className='text-gray-600'>{reply.text}</p>
                        </div>
                    </div>
                </div>
            );
        });
    };

    const closeReviewModal = () => {
        setIsOpen(false);
    };

    const toggleReviewInput = () => {
        setIsOpen(true);
    };

    return (
        <div className='p-10 mb-20 sm:flex justify-center'>
            <div
                onClick={() => toggleReviewInput()}
                className='p-6 bg-accent rounded-full h-8 w-8 flex items-center justify-center text-2xl text-white mt-4 shadow-lg cursor-pointer'
            >
                +
            </div>
            <div className='flex flex-wrap space-x-3 justify-center items-start mt-6 border-t border-gray-200/80 pt-6 space-y-4'>
                {comments &&
                    Object.keys(comments).map((commentId) => {
                        const comment = comments[commentId];
                        const replies = reply[commentId];

                        return (
                            <div
                                key={commentId}
                                className='  bg-white h-full rounded-2xl px-10 py-8 shadow-lg hover:shadow-2xl transition duration-500'
                            >
                                <div className='w-14 h-14  bg-accent rounded-full flex items-center justify-center font-bold text-white'>
                                    <Image
                                        className='w-full h-full object-cover rounded-full'
                                        src={userData.profile_pic} // Assuming each comment has a user object with a profilePic field
                                        alt='User'
                                        width={500}
                                        height={500}
                                    />
                                </div>
                                <div className='mt-4 flex items-center space-x-4'>
                                    <div className='text-sm font-semibold'>
                                        {userData.first_name} â€¢{" "}
                                        <span className='font-normal'>
                                            {" "}
                                            5 minutes ago
                                        </span>
                                    </div>
                                </div>
                                <div className='mt-4'>
                                    <div className='flex items-center mt-2'>
                                        <StarRating rating={5} />
                                    </div>

                                    <p className='mt-4 text-md text-gray-600'>
                                        {comment.text}
                                    </p>
                                    <div className='mt-4 flex justify-between items-center'>
                                        <div
                                            onClick={() =>
                                                toggleReplies(commentId)
                                            }
                                            className='p-2  border-b  rounded-md text-sm text-gray-600 hover:border-accent cursor-pointer'
                                        >
                                            {showReplies[commentId]
                                                ? "Hide Replies"
                                                : "Show Replies"}
                                        </div>
                                    </div>

                                    {renderReplies(replies, commentId)}
                                    <div className='mt-4'>
                                        <input
                                            type='text'
                                            placeholder='Write a reply...'
                                            value={replyText[commentId] || ""} // Set value based on the specific comment's reply text
                                            onChange={(e) =>
                                                handleInputChange(e, commentId)
                                            } // Pass commentId to handleInputChange
                                            className='border border-gray-300 rounded-md p-2 w-full'
                                        />
                                        <button
                                            onClick={() => postReply(commentId)}
                                            className='mt-2 bg-blue-500 text-white px-4 py-2 rounded-md hover:bg-blue-600'
                                        >
                                            Reply
                                        </button>
                                    </div>

                                    <InputReview
                                        isOpen={isOpen}
                                        closeReviewModal={closeReviewModal}
                                    />
                                </div>
                            </div>
                        );
                    })}
            </div>
        </div>
    );
};

export default Comment;
