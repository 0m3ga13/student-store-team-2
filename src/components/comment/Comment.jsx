"use client";
import { useTranslations } from "next-intl";
import React, { useState } from "react";

import ProfilePic from "../profilePicture/ProfilePic";

function Comment({ user, comments, reply }) {
    const t = useTranslations("Index");
    const [showReplyInputs, setShowReplyInputs] = useState({});
    const [replyTexts, setReplyTexts] = useState({});

    const toggleReplyInput = (commentId, replyId) => {
        setShowReplyInputs((prev) => ({
            ...prev,
            [commentId]: {
                ...prev[commentId],
                [replyId]: !prev[commentId]?.[replyId],
            },
        }));
    };

    const handleReplyTextChange = (event, commentId, replyId) => {
        setReplyTexts((prev) => {
            const newReplyTexts = { ...prev };

            if (!newReplyTexts[commentId]) {
                newReplyTexts[commentId] = {};
            }

            newReplyTexts[commentId][replyId] = event.target.value;
            return newReplyTexts;
        });
    };

    const handleReplySubmit = (commentId, replyId) => {
        const replyText = replyTexts[commentId]?.[replyId];

        if (replyText) {
            console.log(
                `Reply submitted for commentId ${commentId}, replyId ${replyId}:`,
                replyText
            );

            setReplyTexts((prev) => {
                const newReplyTexts = { ...prev };

                if (!newReplyTexts[commentId]) {
                    newReplyTexts[commentId] = {};
                }

                newReplyTexts[commentId][replyId] = replyText; // Set reply text to the current value
                return newReplyTexts;
            });
        }

        toggleReplyInput(commentId, replyId);
    };

    const renderReplies = (replies, commentId) => {
        if (!replies) return null;

        return Object.keys(replies).map((replyId) => {
            const reply = replies[replyId];
            const isReplyInputVisible = showReplyInputs[commentId]?.[replyId];

            return (
                <div key={replyId} className='ml-20 my-3'>
                    <div className='flex flex-row items-center'>
                        <ProfilePic user={user} />
                        <div className='text-accent mx-3 '>
                            {user.first_name} {user.last_name}
                        </div>
                        {reply.date}
                    </div>
                    <div className='ml-14 border-l-2 reply bg-accent2 rounded-md shadow-inner shadow-md px-3 py-4'>
                        {reply.text}
                    </div>

                    <div className='flex  ml-20'>
                        <button
                            className='mr-2 px-3 py-2 rounded-md flex items-center'
                            onClick={() => toggleReplyInput(commentId, replyId)}
                        >
                            <svg
                                width='40'
                                height='40'
                                viewBox='0 0 40 40'
                                fill='none'
                                xmlns='http://www.w3.org/2000/svg'
                            >
                                <path
                                    d='M33.3335 3.33301H6.66683C4.8335 3.33301 3.3335 4.83301 3.3335 6.66634V26.6663C3.3335 28.4997 4.8335 29.9997 6.66683 29.9997H30.0002L36.6668 36.6663V6.66634C36.6668 4.83301 35.1668 3.33301 33.3335 3.33301ZM33.3335 28.6663L31.3335 26.6663H6.66683V6.66634H33.3335V28.6663Z'
                                    fill='#72AEC8'
                                />
                            </svg>
                            <span className='ml-2'>{t("Respond")}</span>
                        </button>

                        <button className='px-3 py-2 rounded-md flex items-center'>
                            <svg
                                width='40'
                                height='40'
                                viewBox='0 0 40 40'
                                fill='none'
                                xmlns='http://www.w3.org/2000/svg'
                            >
                                <path
                                    d='M30.0002 35.0003H13.3335V13.3337L25.0002 1.66699L27.0835 3.75033C27.2779 3.94477 27.4379 4.20866 27.5635 4.54199C27.6891 4.87533 27.7513 5.19477 27.7502 5.50033V6.08366L25.9168 13.3337H35.0002C35.889 13.3337 36.6668 13.667 37.3335 14.3337C38.0002 15.0003 38.3335 15.7781 38.3335 16.667V20.0003C38.3335 20.1948 38.3129 20.4031 38.2718 20.6253C38.2307 20.8475 38.1679 21.0559 38.0835 21.2503L33.0835 33.0003C32.8335 33.5559 32.4168 34.0281 31.8335 34.417C31.2502 34.8059 30.6391 35.0003 30.0002 35.0003ZM10.0002 13.3337V35.0003H3.3335V13.3337H10.0002Z'
                                    fill='#72AEC8'
                                />
                            </svg>
                            <span className='ml-2'>{t("Like")}</span>
                        </button>
                    </div>

                    {isReplyInputVisible && (
                        <div className='flex items-center ml-20'>
                            <textarea
                                value={replyTexts[commentId]?.[replyId] || ""}
                                onChange={(event) =>
                                    handleReplyTextChange(
                                        event,
                                        commentId,
                                        replyId
                                    )
                                }
                                onKeyUp={(event) => {
                                    if (
                                        event.key === "Enter" &&
                                        !event.shiftKey
                                    ) {
                                        event.preventDefault();
                                        handleReplySubmit(commentId, replyId);
                                    }
                                }}
                                onInput={(event) => {
                                    event.target.style.height = "auto";
                                    event.target.style.height =
                                        event.target.scrollHeight + "px";
                                }}
                                placeholder={t("Type your reply") + "..."}
                                className='rounded-md shadow-md bg-accent2 w-full py-2 pl-3 ml-3 rounded-tr-none rounded-br-none'
                            />

                            <button
                                className='bg-accent mr-3 text-black rounded-md py-5 px-5 rounded-tl-none rounded-bl-none'
                                onClick={() =>
                                    handleReplySubmit(commentId, replyId)
                                }
                            >
                                {t("Send")}
                            </button>
                        </div>
                    )}
                </div>
            );
        });
    };

    return (
        <div className='container mx-auto my-10 ml-40 mr-40 mt-10 pt-10'>
            <h1 className='text-2xl my-5'>{t("Comments")}</h1>
            <div className='rounded-md shadow-md mr-40'>
                <div className='flex flex-row items-center'>
                    <ProfilePic user={user} />
                    <textarea
                        className='rounded-md shadow-md bg-accent2 w-full py-2 pl-3 ml-3 rounded-tr-none rounded-br-none items-center'
                        placeholder={t("Type your comment") + "..."}
                        onKeyDown={(event) => {
                            if (event.key === "Enter" && !event.shiftKey) {
                                event.preventDefault();

                                const textarea = event.target;
                                const caretPosition = textarea.selectionStart;

                                const valueBeforeCaret =
                                    textarea.value.substring(0, caretPosition);
                                const valueAfterCaret =
                                    textarea.value.substring(caretPosition);

                                // Insert a new line at the cursor position
                                textarea.value =
                                    valueBeforeCaret + "\n" + valueAfterCaret;

                                // If you are using state to manage the textarea value, update the state accordingly
                                // Example: setComment(textarea.value);
                            }
                        }}
                    />

                    <button
                        type='button'
                        className='bg-accent mr-3 text-black rounded-md py-5 px-5 rounded-tl-none rounded-bl-none'
                    >
                        {t("Send")}
                    </button>
                </div>

                {Object.keys(comments).map((commentId) => {
                    const comment = comments[commentId];
                    const replies = reply[commentId];

                    return (
                        <div key={commentId} className='ml-5 my-2'>
                            <div className='flex flex-row items-center'>
                                <ProfilePic user={user} />
                                <div className='text-accent mx-3 '>
                                    {user.first_name} {user.last_name}
                                </div>
                                {comment.date}
                            </div>
                            <div className='ml-14 rounded-md shadow-md px-3 py-4'>
                                <div>{comment.text}</div>
                            </div>

                            <div className='flex ml-20 items-center'>
                                <button
                                    className='mr-2 px-3 py-2 rounded-md flex items-center'
                                    onClick={() =>
                                        toggleReplyInput(commentId, commentId)
                                    }
                                >
                                    <svg
                                        width='40'
                                        height='40'
                                        viewBox='0 0 40 40'
                                        fill='none'
                                        xmlns='http://www.w3.org/2000/svg'
                                    >
                                        <path
                                            d='M33.3335 3.33301H6.66683C4.8335 3.33301 3.3335 4.83301 3.3335 6.66634V26.6663C3.3335 28.4997 4.8335 29.9997 6.66683 29.9997H30.0002L36.6668 36.6663V6.66634C36.6668 4.83301 35.1668 3.33301 33.3335 3.33301ZM33.3335 28.6663L31.3335 26.6663H6.66683V6.66634H33.3335V28.6663Z'
                                            fill='#72AEC8'
                                        />
                                    </svg>
                                    <span className='ml-2'>{t("Respond")}</span>
                                </button>

                                <button className='px-3 py-2 rounded-md flex items-center'>
                                    <svg
                                        width='40'
                                        height='40'
                                        viewBox='0 0 40 40'
                                        fill='none'
                                        xmlns='http://www.w3.org/2000/svg'
                                    >
                                        <path
                                            d='M30.0002 35.0003H13.3335V13.3337L25.0002 1.66699L27.0835 3.75033C27.2779 3.94477 27.4379 4.20866 27.5635 4.54199C27.6891 4.87533 27.7513 5.19477 27.7502 5.50033V6.08366L25.9168 13.3337H35.0002C35.889 13.3337 36.6668 13.667 37.3335 14.3337C38.0002 15.0003 38.3335 15.7781 38.3335 16.667V20.0003C38.3335 20.1948 38.3129 20.4031 38.2718 20.6253C38.2307 20.8475 38.1679 21.0559 38.0835 21.2503L33.0835 33.0003C32.8335 33.5559 32.4168 34.0281 31.8335 34.417C31.2502 34.8059 30.6391 35.0003 30.0002 35.0003ZM10.0002 13.3337V35.0003H3.3335V13.3337H10.0002Z'
                                            fill='#72AEC8'
                                        />
                                    </svg>
                                    <span className='ml-2'>{t("Like")}</span>
                                </button>
                                <span>{comment.likes}</span>
                            </div>

                            {showReplyInputs[commentId] && (
                                <div className='flex items-center ml-20'>
                                    <textarea
                                        value={
                                            replyTexts[commentId]?.[
                                                commentId
                                            ] || ""
                                        }
                                        onChange={(event) =>
                                            handleReplyTextChange(
                                                event,
                                                commentId,
                                                commentId
                                            )
                                        }
                                        onKeyUp={(event) => {
                                            if (
                                                event.key === "Enter" &&
                                                !event.shiftKey
                                            ) {
                                                handleReplySubmit(
                                                    commentId,
                                                    commentId
                                                );
                                            }
                                        }}
                                        onInput={(event) => {
                                            event.target.style.height = "auto";
                                            event.target.style.height =
                                                event.target.scrollHeight +
                                                "px";
                                        }}
                                        placeholder={
                                            t("Type your reply") + "..."
                                        }
                                        className='rounded-md shadow-md bg-accent2 w-full py-2 pl-3 ml-3 rounded-tr-none rounded-br-none'
                                    />

                                    <button
                                        className='bg-accent mr-3 text-black rounded-md py-5 px-5 rounded-tl-none rounded-bl-none'
                                        onClick={() =>
                                            handleReplySubmit(
                                                commentId,
                                                commentId
                                            )
                                        }
                                    >
                                        {t("Send")}
                                    </button>
                                </div>
                            )}

                            {renderReplies(replies, commentId)}
                        </div>
                    );
                })}
            </div>
        </div>
    );
}

export default Comment;
