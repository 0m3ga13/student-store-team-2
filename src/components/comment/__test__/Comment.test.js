import { NextIntlClientProvider } from "next-intl";
import renderer from "react-test-renderer";

import Comment from "../Comment";
import messages from "../../../../messages/en.json";

it("render correnctly", () => {
    const tree = renderer
        .create(
            <NextIntlClientProvider locale='en' messages={messages}>
                <Comment />
            </NextIntlClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
