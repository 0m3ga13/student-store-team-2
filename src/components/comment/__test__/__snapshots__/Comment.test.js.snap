// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correnctly 1`] = `
<div
  className="p-10 mb-20 sm:flex justify-center"
>
  <div
    className="p-6 bg-accent rounded-full h-8 w-8 flex items-center justify-center text-2xl text-white mt-4 shadow-lg cursor-pointer"
    onClick={[Function]}
  >
    +
  </div>
  <div
    className="flex flex-wrap space-x-3 justify-center items-start mt-6 border-t border-gray-200/80 pt-6 space-y-4"
  >
    <div
      className="  bg-white h-full rounded-2xl px-10 py-8 shadow-lg hover:shadow-2xl transition duration-500"
    >
      <div
        className="w-14 h-14  bg-accent rounded-full flex items-center justify-center font-bold text-white"
      >
        <img
          alt="User"
          className="w-full h-full object-cover rounded-full"
          data-nimg="1"
          decoding="async"
          height={500}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1600486913747-55e5470d6f40%3Fauto%3Dformat%26fit%3Dcrop%26q%3D80%26w%3D2070%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1080&q=75"
          srcSet="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1600486913747-55e5470d6f40%3Fauto%3Dformat%26fit%3Dcrop%26q%3D80%26w%3D2070%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1600486913747-55e5470d6f40%3Fauto%3Dformat%26fit%3Dcrop%26q%3D80%26w%3D2070%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1080&q=75 2x"
          style={
            Object {
              "color": "transparent",
            }
          }
          width={500}
        />
      </div>
      <div
        className="mt-4 flex items-center space-x-4"
      >
        <div
          className="text-sm font-semibold"
        >
          Mohammed
           •
           
          <span
            className="font-normal"
          >
             
            5 minutes ago
          </span>
        </div>
      </div>
      <div
        className="mt-4"
      >
        <div
          className="flex items-center mt-2"
        >
          <div
            className="flex text-accent"
          >
            <svg
              className="w-6 h-6 cursor-pointer"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
              />
            </svg>
            <svg
              className="w-6 h-6 cursor-pointer"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
              />
            </svg>
            <svg
              className="w-6 h-6 cursor-pointer"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
              />
            </svg>
            <svg
              className="w-6 h-6 cursor-pointer"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
              />
            </svg>
            <svg
              className="w-6 h-6 cursor-pointer"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
              />
            </svg>
          </div>
        </div>
        <p
          className="mt-4 text-md text-gray-600"
        >
          i like that product &lt;3 
        </p>
        <div
          className="mt-4 flex justify-between items-center"
        >
          <div
            className="p-2  border-b  rounded-md text-sm text-gray-600 hover:border-accent cursor-pointer"
            onClick={[Function]}
          >
            Show Replies
          </div>
        </div>
        <div
          className="mt-4"
        >
          <input
            className="border border-gray-300 rounded-md p-2 w-full"
            onChange={[Function]}
            placeholder="Write a reply..."
            type="text"
            value=""
          />
          <button
            className="mt-2 bg-blue-500 text-white px-4 py-2 rounded-md hover:bg-blue-600"
            onClick={[Function]}
          >
            Reply
          </button>
        </div>
      </div>
    </div>
    <div
      className="  bg-white h-full rounded-2xl px-10 py-8 shadow-lg hover:shadow-2xl transition duration-500"
    >
      <div
        className="w-14 h-14  bg-accent rounded-full flex items-center justify-center font-bold text-white"
      >
        <img
          alt="User"
          className="w-full h-full object-cover rounded-full"
          data-nimg="1"
          decoding="async"
          height={500}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1600486913747-55e5470d6f40%3Fauto%3Dformat%26fit%3Dcrop%26q%3D80%26w%3D2070%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1080&q=75"
          srcSet="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1600486913747-55e5470d6f40%3Fauto%3Dformat%26fit%3Dcrop%26q%3D80%26w%3D2070%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1600486913747-55e5470d6f40%3Fauto%3Dformat%26fit%3Dcrop%26q%3D80%26w%3D2070%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1080&q=75 2x"
          style={
            Object {
              "color": "transparent",
            }
          }
          width={500}
        />
      </div>
      <div
        className="mt-4 flex items-center space-x-4"
      >
        <div
          className="text-sm font-semibold"
        >
          Mohammed
           •
           
          <span
            className="font-normal"
          >
             
            5 minutes ago
          </span>
        </div>
      </div>
      <div
        className="mt-4"
      >
        <div
          className="flex items-center mt-2"
        >
          <div
            className="flex text-accent"
          >
            <svg
              className="w-6 h-6 cursor-pointer"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
              />
            </svg>
            <svg
              className="w-6 h-6 cursor-pointer"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
              />
            </svg>
            <svg
              className="w-6 h-6 cursor-pointer"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
              />
            </svg>
            <svg
              className="w-6 h-6 cursor-pointer"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
              />
            </svg>
            <svg
              className="w-6 h-6 cursor-pointer"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
              />
            </svg>
          </div>
        </div>
        <p
          className="mt-4 text-md text-gray-600"
        >
          what's the price
        </p>
        <div
          className="mt-4 flex justify-between items-center"
        >
          <div
            className="p-2  border-b  rounded-md text-sm text-gray-600 hover:border-accent cursor-pointer"
            onClick={[Function]}
          >
            Show Replies
          </div>
        </div>
        <div
          className="mt-4"
        >
          <input
            className="border border-gray-300 rounded-md p-2 w-full"
            onChange={[Function]}
            placeholder="Write a reply..."
            type="text"
            value=""
          />
          <button
            className="mt-2 bg-blue-500 text-white px-4 py-2 rounded-md hover:bg-blue-600"
            onClick={[Function]}
          >
            Reply
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
